### PR TITLE
Remove use of internaldata as much as possible

### PR DESCRIPTION
**Pull request by atoulme**
Map source and sourcetype to exported events.

**Description:** 
Refactors splunkhec exporter to remove use of internaldata and use pdata instead.

### CODE REVIEW

File: exporter/splunkhecexporter/metricdata_to_splunk.go [hunk 1]
```diff
@@ -15,88 +15,152 @@
...
-	// sumOfSquaredDeviation is the sum of squared deviation metric value suffix.
-	sumOfSquaredDeviation = "sum_of_squared_deviation"
-)
-
-var (
-	// infinity bound dimension value is used on all histograms.
-	infinityBoundSFxDimValue = float64ToDimValue(math.Inf(1))
 )
 
 func metricDataToSplunk(logger *zap.Logger, data pdata.Metrics, config *Config) ([]*splunk.Event, int, error) {
```
Comment (tigrannajaryan) [inline exporter/splunkhecexporter/metricdata_to_splunk.go:37]: Is it possible to split this func into smaller funcs to make it more readable? The nesting of fors, switches and ifs is hard to parse.

File: exporter/splunkhecexporter/tracedata_to_splunk.go [hunk 2]
```diff
@@ -24,34 +24,42 @@ import (
...
+			if sourcetypeSet, isSet := rs.Resource().Attributes().Get(splunk.SourcetypeLabel); isSet {
+				sourceType = sourcetypeSet.StringVal()
+			}
+		}
+		for sils := 0; sils < rs.InstrumentationLibrarySpans().Len(); sils++ {
+			ils := rs.InstrumentationLibrarySpans().At(sils)
+			for si := 0; si < ils.Spans().Len(); si++ {
+				span := ils.Spans().At(si)
+				// TODO replace with span JSON object.
+				spanOC := internaldata.TraceDataToOC(data)[i].Spans[(sils+1)*si]
```
Comment (tigrannajaryan) [inline exporter/splunkhecexporter/tracedata_to_splunk.go:51]: TraceDataToOC is an expensive function. Calling it repeatedly just to get a small portion of the data is inefficient. This likely makes things worse than it was before performance wise. I suggest that we either explicitly convert the data to JSON or wait until interndata exposes the necessary function (unclear yet what exactly we need though that can be expressed as a generic functionality).

**Review by tigrannajaryan — CHANGES_REQUESTED** [review]

File: exporter/splunkhecexporter/tracedata_to_splunk.go (hunk 2 above)
Comment (atoulme) [reply exporter/splunkhecexporter/tracedata_to_splunk.go]: Understood - sounds like at the very least I can move the conversion outside the for loop, and have done so.
This changeset can wait for a better way to express spans as JSON, indeed.
It's just easier for a slew of changes I have queued up - I'll document.

Comment (tigrannajaryan) [reply exporter/splunkhecexporter/tracedata_to_splunk.go]: If we keep the `TraceDataToOC` call anywhere (even a single call outside the loop) then we are not really gaining much here, do we? The dependency on the `internaldata` is still here and likely performance-wise it is going to be overall similar. So I am not sure what the benefit of the change really. I may be missing something.

Comment (tigrannajaryan) [reply exporter/splunkhecexporter/tracedata_to_splunk.go]: @atoulme any thoughts on this? Should we table this for now?

Comment (atoulme) [reply exporter/splunkhecexporter/tracedata_to_splunk.go]: We can table this for now.
I wanted it out of the way so we could start working on other aspects, such as passthrough of index attribute. It would also help with e2e testing so we could check we get the same pdata and splunk events between transformations.

**Comment by tigrannajaryan** [comment]
@atoulme ok, as agreed I am closing this PR for now, please fee free to reopen.
